src/widgets/dropdown.cpp

```diff
@@ -352,62 +352,67 @@ void ShowDropDownListAt(Window *w, const DropDownList *list, int selected, int b
 	/* Longest item in the list, if auto_width is enabled */
 	uint max_item_width = 0;
 
-	/* Total length of list */
-	int height = 0;
+	/* Total height of list */
+	uint height = 0;
 
 	for (const DropDownListItem * const *it = list->Begin(); it != list->End(); ++it) {
 		const DropDownListItem *item = *it;
 		height += item->Height(width);
 		if (auto_width) max_item_width = max(max_item_width, item->Width() + 5);
 	}
 
-	/* Check if the status bar is visible, as we don't want to draw over it */
-	int screen_bottom = GetMainViewBottom();
+	/* Scrollbar needed? */
 	bool scroll = false;
 
-	/* Check if the dropdown will fully fit below the widget */
-	if (top + height + 4 >= screen_bottom) {
-		/* If not, check if it will fit above the widget */
-		int screen_top = GetMainViewTop();
-		if (w->top + wi_rect.top > screen_top + height) {
-			top = w->top + wi_rect.top - height - 4;
-		} else {
-			/* If it doesn't fit above the widget, we need to enable a scrollbar... */
-			int avg_height = height / (int)list->Length();
-			scroll = true;
+	/* Is it better to place the dropdown above the widget? */
+	bool above = false;
 
-			/* ... and choose whether to put the list above or below the widget. */
-			bool put_above = false;
-			int available_height = screen_bottom - w->top - wi_rect.bottom;
-			if (w->top + wi_rect.top - screen_top > available_height) {
-				// Put it above.
-				available_height = w->top + wi_rect.top - screen_top;
-				put_above = true;
-			}
+	/* Available height below (or above, if the dropdown is placed above the widget). */
+	uint available_height = (uint)max(GetMainViewBottom() - top - 4, 0);
```

Comment:
i feel like `max<uint>` would be cleaner - no need to cast after the function call (+ available_height_above below)